### PR TITLE
par: fix static build

### PR DESCRIPTION
**Pull request by KAction**
- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/par/default.nix [hunk 1]
```diff
@@ -18,7 +22,7 @@ stdenv.mkDerivation {
     })
   ];
 
-  buildPhase = ''make -f protoMakefile'';
+  buildPhase = ''make -f protoMakefile CC="${cc} -c" LINK1=${cc}'';
```
Comment (Mic92) [inline pkgs/tools/text/par/default.nix:25]: targetPrefix should be empty for non-cross builds.

```suggestion
  buildPhase = ''make -f protoMakefile CC="${stdenv.cc.targetPrefix}cc -c" LINK1=${stdenv.cc.targetPrefix}cc'';
```

File: pkgs/tools/text/par/default.nix [hunk 2]
```diff
@@ -18,7 +18,11 @@ stdenv.mkDerivation {
     })
   ];
 
-  buildPhase = ''make -f protoMakefile'';
+  buildPhase = ''
+    make -f protoMakefile \
+      CC="${stdenv.cc.targetPrefix}cc -c" \
+      LINK1=${stdenv.cc.targetPrefix}cc
+  '';
```
Comment (Mic92) [inline pkgs/tools/text/par/default.nix:25]: I did not saw this earlier, but should it not be feasible to just use `makeFlags` here?
 
makeFlags = [ 
  "-f" "protoMakefile"
  "CC=${stdenv.cc.targetPrefix}cc -c"
  "LINK1=${stdenv.cc.targetPrefix}cc"
];

Comment (KAction) [reply pkgs/tools/text/par/default.nix:25]: Does not work for me due space inside of CC. -c gets interpreted as make(1) flag.

Comment (KAction) [reply pkgs/tools/text/par/default.nix]: But can work with makeFlagsArray. Do you like this approach (just pushed) better?

File: pkgs/tools/text/par/default.nix [hunk 3]
```diff
@@ -18,7 +18,12 @@ stdenv.mkDerivation {
     })
   ];
 
-  buildPhase = ''make -f protoMakefile'';
+  makefile = "protoMakefile";
+  preBuild = ''
+    makeFlagsArray=( CC="${stdenv.cc.targetPrefix}cc -c" \
```
Comment (Mic92) [inline pkgs/tools/text/par/default.nix:23]: This allows to still add stuff to makeFlagsArray in an override:

```suggestion
    makeFlagsArray+=( CC="${stdenv.cc.targetPrefix}cc -c" \
```

Comment (KAction) [reply pkgs/tools/text/par/default.nix]: Ok, force-pushed.

**Comment by Mic92** [comment]
@GrahamcOfBorg build par

**Pull request merged**